### DISCUSSION TITLE
Why does submit perform a fetch

### DISCUSSION THREAD

**Discussion by jakejx** [arxanas/git-branchless#671] [Q&A] [answered]
First of, thanks for this great tool, it has really improved by Git workflow significantly.

I was trying out the `git submit` to push my stack and I noticed it tried to perform a `git fetch origin` first. At my co, all developer share the same repo and this caused git to start fetching all remote branches (and `git-branchless` had to process each one).

This takes quite abit of time so I am wondering:
1. Is it necessary to fetch all the remote branches? or
2. Is there a way to reduce the branches fetched?

Thanks!

**Comment by arxanas** [comment, marked as answer]
Hi @jakejx, thanks for reporting. This is something I didn't implement rather than a feature. The reason that we do a fetch at all is to determine if the branch was updated as part of the submit operation or not, but this only requires that we fetch your remote branches. If you're interested, I can show you where in the code you could make the changes to only fetch your relevant branches.

For the general Git mechanism to reduce fetched branches, you should use a [refspec](https://git-scm.com/book/en/v2/Git-Internals-The-Refspec) and specify that you only want to fetch the repository main branch(es) and possibly your own branches. You can always manually fetch a given branch by name later before switching to it.

**Reply by jakejx** [reply]
Hey, sorry for getting back after so long. I have found the location that performs the fetch, but I haven't had the chance to work on a code change. I'll see if I have a chance to work on this.
